src/client/dfs/dfs.c

```diff
@@ -2147,16 +2147,69 @@ dfs_disconnect(dfs_t *dfs)
 	return rc;
 }
 
+static int
+get_uid_gid_from_ownership(daos_handle_t coh, dfs_t *dfs)
+{
+	daos_prop_t		*owner_props;
+	struct daos_prop_entry	*entry;
+	int			rc = 0;
+
+	owner_props = daos_prop_alloc(2);
+	if (owner_props == NULL)
+		return ENOMEM;
+
+	owner_props->dpp_entries[0].dpe_type = DAOS_PROP_CO_OWNER;
+	owner_props->dpp_entries[1].dpe_type = DAOS_PROP_CO_OWNER_GROUP;
+
+	rc = daos_cont_query(coh, NULL, owner_props, NULL);
```

Comment:
instead of calling daos_cont_query twice, why not call it once and depending on the dfs layout, call it with three props or 5?

Reply:
I wondered this, you don't know if you need the extra two until you've used the first set do you?

Reply:
ah yes indeed, since we have to get the SB oid to be able to query the layout from there. i missed that part.
so querying non existing props returns errors?

Reply:
i understand that it's better to query only the props that are needed and we should update this. that definitely is better. 
bit i don't understand though why querying all properties returns an error depending on the ACL. Unless we have ACLs that restrict querying some properties but not others, shouldn't querying 3 properties return the same error as querying all properties if there are ACL permission issues?

Reply:
> Calling daos_cont_query() twice was my idea to streamline the normal case although it wouldn't be wrong to query all five up-front.

on that front it's cleaner to me to just query all 5 to reduce query calls, prop alloc, prop frees.

Reply:
We have ACLs that restrict querying some properties but not others, but none of the affected properties are used by dfs.

Reply:
> > Calling daos_cont_query() twice was my idea to streamline the normal case although it wouldn't be wrong to query all five up-front.
> 
> on that front it's cleaner to me to just query all 5 to reduce query calls, prop alloc, prop frees.

but i'll retract my request for change there as indeed querying 3 is the normal case, and we shouldn't call query twice unless we are reading 2.0 container which i don't anticipate would happen much

Reply:
Will fix.